src/compute-client/src/controller.rs

```diff
@@ -489,6 +489,18 @@ where
         Ok(())
     }
 
+    /// Drop the read capability for the given collections and allow their resources to be
+    /// reclaimed.
```

Comment:
Thinking more about it, I think this is actually a bug in the current implementation. For each storage input to a dataflow, the controller installs a read hold for each active replica. It advance these read holds when the replicas report `FrontierUpper`s and drops them as replicas are dropped or advance to the empty frontier.

But if we just drop our collection state these read holds leak and there is nothing that cleans them up again. So in `force_drop_collections` we would have to at least remove the per-replica read holds too. Unfortunately, there is a reason why these read holds exist and removing them without feedback from the replica risks that replicas start panicking because they try to read from compacted persist shards.